onie-mk-demo.sh

```diff
@@ -130,7 +133,43 @@ cp $installer_dir/sharch_body.sh $output_file || {
 # Replace variables in the sharch template
 sed -i -e "s/%%IMAGE_SHA1%%/$sha1/" $output_file
 echo -n "."
+tar_size="$(wc -c < "${sharch}")"
+sed -i -e "s|%%PAYLOAD_IMAGE_SIZE%%|${tar_size}|" ${output_file}
 cat $sharch >> $output_file
+echo "secure upgrade flags: SECURE_UPGRADE_MODE = $SECURE_UPGRADE_MODE, \
+SECURE_UPGRADE_DEV_SIGNING_KEY = $SECURE_UPGRADE_DEV_SIGNING_KEY, SECURE_UPGRADE_DEV_SIGNING_CERT = $SECURE_UPGRADE_DEV_SIGNING_CERT"
+
+if [ "$SECURE_UPGRADE_MODE" = "dev" -o "$SECURE_UPGRADE_MODE" = "prod" ]; then
+    CMS_SIG="${tmp_dir}/signature.sig"
+    echo "$0 Creating CMS signature for ${output_file} with  ${key_file}. Output file ${CMS_SIG}"
+    DIR="$(dirname "$0")"
+    scripts_dir="${DIR}/scripts"
+    if [ "$SECURE_UPGRADE_MODE" = "dev" ]; then
+        . ${scripts_dir}/sign_image_dev.sh
+        sign_image_dev ${cert_file} ${key_file} ${output_file} ${CMS_SIG} || {
+        echo "CMS sign error $?"
+        sudo rm -rf ${CMS_SIG}
+        clean_up 1
+    }
+    else # "$SECURE_UPGRADE_MODE" has to be equal to "prod"
+        . ${scripts_dir}/sign_image_${platform}.sh
```

Comment:
If the script does not exists in that location it fails, but after retry it will mark the build as successful:


> 18:22:33  + . ./scripts/sign_image_x86_64-mellanox-r0.sh
> 18:22:33  ./onie-mk-demo.sh: 155: .: cannot open ./scripts/sign_image_x86_64-mellanox-r0.sh: No such file
> 18:22:33  [  FAIL LOG END  ] [ target/sonic-mellanox.bin ]
> 18:22:33  make: *** [slave.mk:1217: target/sonic-mellanox.bin] Error 1
> 18:22:55  Makefile.work:568: recipe for target 'target/sonic-mellanox.bin' failed
> 18:22:55  make[1]: *** [target/sonic-mellanox.bin] Error 2
> 18:22:55  make[1]: Leaving directory '/builds2/sw-r2d2-bot/workspace/sonic_build@2/sonic'
> 18:22:55  Makefile:39: recipe for target 'target/sonic-mellanox.bin' failed
> 18:22:55  make: *** [target/sonic-mellanox.bin] Error 2

And after a rebuild:
18:23:16  make: 'target/sonic-mellanox.bin' is up to date.